File: Dragon_Dungeons.client/src/models/Character.js
import { Profile } from "./Account.js"
import { RepoItem } from "./RepoItem.js"

export class Character extends RepoItem {
  constructor(data) {
    super(data)
    this.name = data.name
    this.picture = data.picture
    this.hp = data.hp
    this.maxHp = data.maxHp
    this.tempHp = data.tempHp
    this.speed = data.speed
    this.hitDie = data.hitDie
    this.level = data.level
    this.class = data.class
    this.race = data.race
    this.alignment = data.alignment
    this.age = data.age
    this.feet = data.feet
    this.inches = data.inches
    this.weight = data.weight
    this.eyes = data.eyes
    this.skin = data.skin
    this.hair = data.hair
    this.features = data.features
    this.background = data.background
    this.backstory = data.backstory
    this.personalityTraits = data.personalityTraits
    this.ideals = data.ideals
    this.bonds = data.bonds
    this.flaws = data.flaws
    this.manual = data.manual
    this.str = data.str
    this.dex = data.dex
    this.con = data.con
    this.int = data.intelligence || data.int
    this.wis = data.wis
    this.cha = data.cha
    this.bonus = data.bonus
    this.skills = data.skills
    this.proficiencies = data.proficiencies
    this.cantrips = data.cantrips
    this.spells = data.spells
    this.equipment = data.equipment
    this.armor = data.armor
    this.weapons = data.weapons || []
    this.creator = new Profile(data.creator)
    this.creatorId = data.creatorId
  }
}

File: Dragon_Dungeons/Models/Character.cs
namespace Dragon_Dungeons.Models
{
  public class Character : RepoItem
  {
    public string Name { get; set; }
    public string Picture { get; set; }
    public int Hp { get; set; }
    public int MaxHp { get; set; }
    public int TempHp { get; set; }
    public int Speed { get; set; }
    public int HitDie { get; set; }
    public int Level { get; set; }
    public string Class { get; set; }
    public string Race { get; set; }
    public string Alignment { get; set; }
    public int Age { get; set; }
    public int Feet { get; set; }
    public int Inches { get; set; }
    public int Weight { get; set; }
    public string Eyes { get; set; }
    public string Skin { get; set; }
    public string Hair { get; set; }
    public string Features { get; set; }
    public string Background { get; set; }
    public string Backstory { get; set; }
    public string PersonalityTraits { get; set; }
    public string Ideals { get; set; }
    public string Bonds { get; set; }
    public string Flaws { get; set; }
    public bool Manual { get; set; }
    public int Str { get; set; }
    public int Dex { get; set; }
    public int Con { get; set; }
    public int Intelligence { get; set; }
    public int Wis { get; set; }
    public int Cha { get; set; }
    public int Bonus { get; set; }
    public string Skills { get; set; }
    public string Proficiencies { get; set; }
    public string Cantrips { get; set; }
    public string Spells { get; set; }
    public string Equipment { get; set; }
    public string CreatorId { get; set; }
    public Profile Creator { get; set; }
  }
}